README.md

```diff
@@ -1,15 +1,174 @@
+        YES --> @TrackBall + time:10
+        NO --> @SearchBall
+    PATTERN --> @LookAround
+```
+
+## Subtrees
+
+To avoid duplication and deep nested structures in the DSD file, subtrees can be used.
+Here is an example that is equivalent to the example above:
+```
+#BallMode
+$BallSeen
+    YES --> @TrackBall
+    NO --> @SearchBall
+
+-->
+$Role
+    BALL --> #BallMode
+    PATTERN --> @LookAround
+```
+
+Subtrees can also receive parameters.
+Here is another example:
+```
+
+#BallMode
+$BallSeen + tracktime
+    YES --> @TrackBall + time:*tracktime
+    NO --> @SearchBall
+
+-->
+$Role
+    BALL --> #BallMode + tracktime:10
+    PATTERN --> @LookAround
+```
+
+As you can see, the parameters are specified in the same way as for actions or decisions.
+In the subtree, the parameter has to be given a name using the `+` syntax and can be referenced with the `*` operator.
+
+## ELSE
+
+`ELSE` is a catch-all decision result.
+It can be used to map multiple results to a single action or decision.
+For example, the `Role` decision could return the roles `BALL`, `GOAL`, `ROBOT`, and `PATTERN`, but the latter three should result in the same behavior:
+```
+-->
+$Mode
+    BALL --> $BallSeen
+        YES --> @TrackBall
+        NO --> @SearchBall
+    ELSE --> @LookAround
+```
+
+## Interrupt
+
+A stack element can call `self.interrupt()` to create an interrupt which removes all elements except for the start element from the stack (i.e., the DSD is restarted). 
+
+## Debugging
+
+For debugging, the DSD visualization should be used.
+It reads the locations of different DSDs from `config/locations.yaml` and can be launched in rqt.
+The visualization shows the DSD tree with the currently active actions and decisions on the left and the current stack on the right.
+More information can be added to the right side by calling `self.publish_debug_data(label, data)` in any action or decision.
+
+## Examples
+
+Here are a few projects that use the DSD and can be used for reference:
+* The [Bit-Bots Head Behavior](https://github.com/bit-bots/bitbots_behavior/tree/master/bitbots_head_behavior/src/bitbots_head_behavior)
+* The [Bit-Bots Body Behavior](https://github.com/bit-bots/bitbots_behavior/tree/master/bitbots_body_behavior/src/bitbots_body_behavior)
+* The [Bit-Bots Humanoid Control Module](https://github.com/bit-bots/bitbots_motion/tree/master/bitbots_hcm/src/bitbots_hcm/hcm_dsd)
+* The [Parser Test](https://github.com/bit-bots/dynamic_stack_decider/tree/master/dynamic_stack_decider/tests) in this repository
+
+## The Paper
+
+The framework is also described in more depth in [ASDS – Active Self Deciding Stack: A Lightweight Approach for Dynamic Decision Making](https://www.researchgate.net/profile/Martin_Poppinga/publication/327509965_ASDS_-_Active_Self_Deciding_Stack_A_Lightweight_Approach_for_Dynamic_Decision_Making/links/5b92824b4585153a5301618e/ASDS-Active-Self-Deciding-Stack-A-Lightweight-Approach-for-Dynamic-Decision-Making.pdf)
```

Comment:
we have a newer better paper
https://www.researchgate.net/publication/350168815_DSD_-_Dynamic_Stack_Decider_A_Lightweight_Decision_Making_Framework_for_Robots_and_Software_Agents